### PR TITLE
PReLU with element-wise scales

### PR DESCRIPTION
**Pull request by dkurt**
### Pull Request Readiness Checklist

resolves https://github.com/opencv/opencv/issues/24051

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

```
force_builders=Custom

Xbuild_image:Custom=ubuntu-openvino-2021.4.2:20.04
build_image:Custom=ubuntu-openvino-2022.1.0:20.04

test_modules:Custom=dnn,python2,python3,java,gapi,video

buildworker:Custom=linux-1
# disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=ON
test_bigdata:Custom=1
test_filter:Custom=*
```

### CODE REVIEW

**Comment by asmorkalov** [comment]
```
[ RUN      ] Test_Caffe_nets.FasterRCNN_vgg16/2, where GetParam() = NGRAPH/OCL_FP16
[ INFO:0@130.032] global net_openvino.cpp:676 switchToOpenVINOBackend DNN: switching to OpenVINO backend... (networkID=450)
[ INFO:0@130.342] global ie_ngraph.cpp:823 initPlugin OpenCV/nGraph: using GPU kernels cache: /build/.cache/openvino_cache_GPU_ubuntu_openvino_2022.1.0__20.04/
[ INFO:0@130.343] global ie_ngraph.cpp:860 initPlugin DNN/IE: Calling LoadNetwork(device=HETERO:GPU,CPU)...
Unmatched prediction: class 2 score 0.964355 box [473.766 x 378.733 from (108.551, 130.3)]
Highest IoU: 0.641764
/build/precommit_custom_linux/4.x/opencv/modules/dnn/test/test_common.impl.hpp:145: Failure
Value of: matched
  Actual: false
Expected: true
model name: VGG16_faster_rcnn_final.caffemodel
Unmatched reference: class 2 score 0.949398 box [501.96 x 252.708 from (99.2454, 210.141)] IoU diff: 0.358236
/build/precommit_custom_linux/4.x/opencv/modules/dnn/test/test_common.impl.hpp:157: Failure
Expected: (refScores[i]) <= (confThreshold), actual: 0.949398 vs 0.8
model name: VGG16_faster_rcnn_final.caffemodel
[ INFO:0@132.404] global ts.cpp:857 testTearDown Memory_usage (OpenCL): 489242624 (base=0  current=0)
[  FAILED  ] Test_Caffe_nets.FasterRCNN_vgg16/2, where GetParam() = NGRAPH/OCL_FP16 (2956 ms)
```

**Comment by dkurt** [comment]
@asmorkalov, this error is reproduced on 4.x:
```
[ RUN      ] Test_Caffe_nets.FasterRCNN_vgg16/2, where GetParam() = NGRAPH/OCL_FP16
Unmatched prediction: class 2 score 0.910156 box [528.531 x 252.169 from (77.7105, 209.179)]
Highest IoU: 0
/home/dkurt/opencv/modules/dnn/test/test_common.impl.hpp:145: Failure
Value of: matched
  Actual: false
Expected: true
model name: VGG16_faster_rcnn_final.caffemodel
Unmatched reference: class 2 score 0.949398 box [501.96 x 252.708 from (99.2454, 210.141)] IoU diff: 1
/home/dkurt/opencv/modules/dnn/test/test_common.impl.hpp:157: Failure
Expected: (refScores[i]) <= (confThreshold), actual: 0.949398 vs 0.8
model name: VGG16_faster_rcnn_final.caffemodel
[  FAILED  ] Test_Caffe_nets.FasterRCNN_vgg16/2, where GetParam() = NGRAPH/OCL_FP16 (13016 ms)
```

File: modules/dnn/src/opencl/activations.cl [hunk 1]
```diff
@@ -73,14 +73,23 @@ __kernel void ReLU6Forward(const int count, __global const T* in, __global T* ou
...
+
 __kernel void PReLUForward(const int count, const int channels, const int plane_size,
                            __global const T* in, __global T* out,
                            __global const KERNEL_ARG_DTYPE* slope_data)
 {
   int index = get_global_id(0);
-  int c = (index / plane_size) % channels;
   if(index < count)
-  out[index] = in[index] > 0 ? in[index] : in[index] * slope_data[c];
+  out[index] = in[index] > 0 ? in[index] : in[index] * slope_data[index];
```
Comment (asmorkalov) [inline modules/dnn/src/opencl/activations.cl:92]: formatting issue.

File: modules/dnn/src/opencl/activations.cl [hunk 2]
```diff
@@ -73,14 +73,23 @@ __kernel void ReLU6Forward(const int count, __global const T* in, __global T* ou
...
 }
 
+__kernel void ChannelsPReLUForward(const int count, const int channels, const int plane_size,
+                                   __global const T* in, __global T* out,
+                                   __global const KERNEL_ARG_DTYPE* slope_data)
+{
+  int index = get_global_id(0);
+  int c = (index / plane_size) % channels;
+  if(index < count)
+  out[index] = in[index] > 0 ? in[index] : in[index] * slope_data[c];
```
Comment (asmorkalov) [inline modules/dnn/src/opencl/activations.cl:83]: formatting issue

File: modules/dnn/src/layers/elementwise_layers.cpp [hunk 3]
```diff
@@ -355,7 +355,7 @@ struct ReLUFunctor : public BaseFunctor
                backendId == DNN_BACKEND_CANN;
     }
 
-    void apply(const float* srcptr, float* dstptr, int len, size_t planeSize, int cn0, int cn1) const
+    void apply(const float* srcptr, float* dstptr, int, int len, size_t planeSize, int cn0, int cn1) const
```
Comment (asmorkalov) [inline modules/dnn/src/layers/elementwise_layers.cpp:358]: I propose to restore new parameter name and use `CV_UNUSED()` for it. It makes code cleaner for readers.

**Comment by Abdurrahheem** [comment]
Look Good to me!

**Review by Abdurrahheem — COMMENTED** [review]
LGTM

**Pull request merged**